src/main/java/com/mineinabyss/mobzy/api/nms/entity/ProjectileHelpers.kt

```diff
@@ -0,0 +1,47 @@
+package com.mineinabyss.mobzy.api.nms.entity
+
+import net.minecraft.server.v1_16_R2.Entity.c
+import net.minecraft.server.v1_16_R2.IProjectile
+import net.minecraft.server.v1_16_R2.MathHelper
+import net.minecraft.server.v1_16_R2.Vec3D
+import org.bukkit.util.Vector
+import kotlin.math.atan2
+import kotlin.math.sqrt
+import kotlin.random.Random
+
+
+/**
+ * Shoot the projectile towards a direction, with a certain speed and randomness
+ * @param dX X direction
+ * @param dY Y direction
+ * @param dZ Z direction
+ * @param speed The speed at which the projectile should travel
+ * @param randomAngle The maximum random angle that can be applied to the projectile shooting direction. Set to 0 to aim directly at the target.
+ */
+//TODO: Implement the entity deltaPosition ticking ourselves at some point, to circumvent NMS.
+fun IProjectile.shootDirection(dX: Double, dY: Double, dZ: Double, speed: Float, randomAngle: Double) {
```

Comment:
I'm fine with leaving this here, we might get more helpers in the future and it seems useful enough to others to be in the api package (which might be split off into its own thing in the future)